packages/block/src/header.ts

```diff
@@ -238,6 +240,11 @@ export class BlockHeader {
       this.difficulty = this.canonicalDifficulty(options.calcDifficultyFromHeader)
     }
 
+    // If cliqueSigner is provided, seal block with provided privateKey.
+    if (options.cliqueSigner) {
+      this.extraData = this.cliqueSealBlock(options.cliqueSigner)
+    }
```

Comment:
Nice, also like that the private key is not residing in the library, was a bit worried that we introduce some new potential security risks here.